run.go

```diff
@@ -67,7 +67,8 @@ func runflags() []cli.Flag {
 	}
 }
 
-func preRun(c *cli.Context) (*funcfile, []string, error) {
+// preRun parses func.yaml, checks expected env vars and builds the function image.
+func preRun(c *cli.Context) (string, *funcfile, []string, error) {
```

Comment:
nit: probably time to return an object here, without named params this is gobbledy